src/reactviews/pages/SchemaCompare/SchemaCompareStateProvider.tsx

```diff
@@ -0,0 +1,149 @@
+ *--------------------------------------------------------------------------------------------*/
+
+import * as sc from "../../../sharedInterfaces/schemaCompare";
+import * as mssql from "vscode-mssql";
+
+import { createContext } from "react";
+import { useVscodeWebview } from "../../common/vscodeWebviewProvider";
+
+const schemaCompareContext = createContext<sc.SchemaCompareContextProps>(
+    {} as sc.SchemaCompareContextProps,
+);
+
+interface SchemaCompareStateProviderProps {
+    children: React.ReactNode;
+}
+
+const SchemaCompareStateProvider: React.FC<SchemaCompareStateProviderProps> = ({
+    children,
+}) => {
+    const webViewState = useVscodeWebview<
+        sc.SchemaCompareWebViewState,
+        sc.SchemaCompareReducers
+    >();
+    const schemaCompareState = webViewState?.state;
+
+    return (
+        <schemaCompareContext.Provider
+            value={{
+                state: schemaCompareState,
+                themeKind: webViewState?.themeKind,
+                schemaCompare: function (
+                    operationId: string,
+                    sourceEndpointInfo: mssql.SchemaCompareEndpointInfo,
+                    targetEndpointInfo: mssql.SchemaCompareEndpointInfo,
+                    taskExecutionMode: mssql.TaskExecutionMode,
+                    deploymentOptions: mssql.DeploymentOptions,
+                ): void {
+                    webViewState?.extensionRpc.action("schemaCompare", {
+                        operationId: operationId,
+                        sourceEndpointInfo: sourceEndpointInfo,
+                        targetEndpointInfo: targetEndpointInfo,
+                        taskExecutionMode: taskExecutionMode,
+                        deploymentOptions: deploymentOptions,
+                    });
+                },
+                schemaCompareGenerateScript: function (
+                    operationId: string,
+                    targetServerName: string,
+                    targetDatabaseName: string,
+                    taskExecutionMode: mssql.TaskExecutionMode,
+                ): void {
+                    webViewState?.extensionRpc.action(
+                        "schemaCompareGenerateScript",
+                        {
+                            operationId: operationId,
+                            targetServerName: targetServerName,
+                            targetDatabaseName: targetDatabaseName,
+                            taskExecutionMode: taskExecutionMode,
+                        },
+                    );
+                },
+                schemaComparePublishDatabaseChanges: function (
+                    operationId: string,
+                    targetServerName: string,
+                    targetDatabaseName: string,
+                    taskExecutionMode: mssql.TaskExecutionMode,
+                ): void {
+                    webViewState?.extensionRpc.action(
+                        "schemaComparePublishDatabaseChanges",
```

Comment:
the two (or three if you add dacpac) publish targets can probably be combined, just with adding an enum for targetType.  it'd be cleaner to split these up in the controller based on the enum, IMO